src/main/java/org/elasticsearch/index/translog/TranslogSnapshot.java

```diff
@@ -70,13 +68,17 @@ public int estimatedTotalOperations() {
     public Translog.Operation next() throws IOException {
         ensureOpen();
         for (; currentTranslog < orderedTranslogs.size(); currentTranslog++) {
-            final ChannelSnapshot current = orderedTranslogs.get(currentTranslog);
+            final Translog.Snapshot current = orderedTranslogs.get(currentTranslog);
             Translog.Operation op = null;
             try {
-                op = current.next(cacheBuffer);
+                op = current.next();
             } catch (TruncatedTranslogException e) {
-                // file is empty or header has been half-written and should be ignored
-                logger.trace("ignoring truncation exception, the translog [{}] is either empty or half-written", e, current.translogId());
+                if (estimatedTotalOperations == TranslogReader.UNKNOWN_OP_COUNT) {
```

Comment:
Hmm, but in LegacyTranslogReaderBase.newReaderSnapshot, it returns null once you've iterated beyond sizeInBytes()?  So can it ever (legitimately) throw TruncatedTranslogExc?  If so, can we maybe add an assert to be sure this is really only happening in the narrow case that we expect?